cpp/src/arrow/array/array_test.cc

```diff
@@ -388,6 +388,7 @@ static std::vector<std::shared_ptr<DataType>> TestArrayUtilitiesAgainstTheseType
       large_utf8(),
       list(utf8()),
       list(int64()),  // NOTE: Regression case for ARROW-9071/MakeArrayOfNull
+      list(large_utf8()),
```

Comment:
Can we also add some nested list types into the mix?

Reply:
I added a few more nested combinations